WordPress/src/test/java/org/wordpress/android/ui/sitecreation/MainVM/NewSiteCreationMainVMTest.kt

```diff
@@ -0,0 +1,209 @@
+        val state = SiteCreationCompleted(LOCAL_SITE_ID)
+        viewModel.onSitePreviewScreenFinished(state)
+
+        val captor = ArgumentCaptor.forClass(CreateSiteState::class.java)
+        verify(wizardFinishedObserver).onChanged(captor.capture())
+
+        assertThat(captor.value).isEqualTo(state)
+    }
+
+    @Test
+    fun nextStepAfterOnBackPressedIsCurrentStep() {
+        val startingStep = currentStep()
+        viewModel.onBackPressed() // navigate to previous step
+        viewModel.onSkipClicked() // navigate to next step
+        assertThat(currentStep()).isEqualTo(startingStep)
+    }
+
+    @Test
+    fun backSupressedOnlyForLastStep() {
+        while (!isAtLastStep()) {
+            assertThat(viewModel.shouldSuppressBackPress()).isFalse()
+            viewModel.onSkipClicked() // navigate to a next step
+        }
+        assertThat(viewModel.shouldSuppressBackPress()).isTrue()
+    }
+
+    @Test
+    fun titleForFirstStepIsGeneralSiteCreation() {
+        assertThat(viewModel.screenTitleForWizardStep(SITE_CREATION_STEPS.first()))
+                .isInstanceOf(ScreenTitleGeneral::class.java)
+    }
+
+    @Test
+    fun titleForLastStepIsEmptyTitle() {
+        assertThat(viewModel.screenTitleForWizardStep(SITE_CREATION_STEPS.last()))
+                .isInstanceOf(ScreenTitleEmpty::class.java)
+    }
+
+    @Test
+    fun titlesForOtherThanFirstAndLastStepIsStepCount() {
+        val isFirstStep = { index: Int -> index == 0 }
+        val isLastStep = { index: Int -> index == SITE_CREATION_STEPS.size - 1 }
+
+        SITE_CREATION_STEPS.filterIndexed({ index, step ->
+            !isFirstStep(index) && !isLastStep(index)
+        }).forEach() { step ->
+            assertThat(viewModel.screenTitleForWizardStep(step)).isInstanceOf(ScreenTitleStepCount::class.java)
+        }
+    }
+
+    @Test
+    fun siteCreationStateWrittenToBundle() {
+        viewModel.writeToBundle(savedInstanceState)
+        verify(savedInstanceState).putParcelable(any(), argThat() { this is SiteCreationState })
+    }
+
+    @Test
+    fun siteCreationStateRestored() {
+        val expectedState = SiteCreationState()
+        whenever(savedInstanceState.getParcelable<SiteCreationState>(any()))
+                .thenReturn(expectedState)
+
+        // we need to create a new instance of the VM as the `viewModel` has already been started in setUp()
+        val newViewModel = NewSiteCreationMainVM(stepsProvider, tracker)
+        newViewModel.start(savedInstanceState)
+
+        /* we need to navigate to the next step as the value of navigationTargetObservable isn't changed when the VM
+        is restored from a savedInstanceState. */
+        newViewModel.onSkipClicked()
+
+        newViewModel.navigationTargetObservable.observeForever(navigationTargetObserver)
+        assertThat(newViewModel.navigationTargetObservable.lastEvent!!.wizardState).isSameAs(expectedState)
+    }
+
+    private fun verifyNavigatedToNextStep() {
```

Comment:
I feel like this name doesn't really reflect what this method does. It's actually checking both the `start` method and skip or `onSelected` actions triggering the navigation. Furthermore, I think we are forcing all the tests to have the knowledge of skipping the first event in their verification. How about we ignore the first event in the `setup` phase instead and keep the tests clean?

Something like this seems to work:

```
// TODO: Put this either outside of the test class or in a helper file as a public method instead
private fun <T> SingleEventObservable<T>.ignoreFirstEventThenObserveForever(observer: Observer<T>) {
    var shouldIgnore = true
    observeForever(Observer {
        if (shouldIgnore) {
            shouldIgnore = false
            return@Observer
        }
        observer.onChanged(it)
    })
}

@Before
fun setUp() {
    // Removed other stuff to keep this brief
    viewModel.navigationTargetObservable.ignoreFirstEventThenObserveForever(navigationTargetObserver)
}
```

I am sure there is a better way to write the `ignoreFirstEventThenObserveForever` function, but I can't think straight right now from the cold 😰